src/v/net/probes.cc

```diff
@@ -282,4 +299,187 @@ std::ostream& operator<<(std::ostream& o, const client_probe& p) {
       << ", requests_blocked_memory: " << p._requests_blocked_memory << " }";
     return o;
 }
+
+ss::future<ss::shared_ptr<ss::tls::server_credentials>>
+build_reloadable_server_credentials_with_probe(
+  config::tls_config config,
+  std::string_view service,
+  std::string_view listener_name,
```

Comment:
Should we avoid `std::string_view` as a parameter to a coroutine? The next commit calls this under a thread, but is this safe in the general case?